src/lib/dataset/mongo.ts

```diff
@@ -67,3 +67,38 @@ export function _guessType(val: any, prevType: any = null): DataSetColumnType |
 
   return null;
 }
+
+/**
+ * extend a `DataSet` headers with inferred types, base on value of columns
+ */
+export function inferTypeFromDataset(dataset: DataSet, maxRows: number = 50): DataSet {
+  // For the moment, we infer arbitrarily column's types on maximum the 50 first rows
+  maxRows = Math.min(dataset.data.length, maxRows);
+  const newHeaders: DataSetColumn[] = [];
+
+  dataset.headers.forEach((header, i) => {
+    let prevType = null;
+
+    for (let j = 0; j < maxRows; j++) {
```

Comment:
```suggestion
for (const rowData of dataset.data) {
  const currentValue = rowData[colIdx];  // ← assuming that you've renamed `i` into `colIdx`
```

Reply:
How do you propose to handle `maxRows` now ?

Reply:
Ah yes, your're right! Calling `.slice(maxRows)` would be a solution I guess, although less optimal computationally speaking.